# U.S. Medical Insurance Costs

Import the dataset with using pandas

In [39]:
#import pandas
import pandas as pd

mi = pd.read_csv("insurance.csv")


What is the % of the people smoke in the dataset?

In [6]:
# Calculate the percentage of smokers in the dataset
smoker_percentage = (mi['smoker'] == 'yes').mean() * 100

smoker_percentage


20.47832585949178

Compare the percentage of male and female smokers

In [24]:
male_smoker_percentage = (mi.loc[mi['sex'] == 'male', 'smoker'] == 'yes').mean() * 100
female_smoker_percentage = (mi.loc[mi['sex'] == 'female', 'smoker'] == 'yes').mean() * 100

print(male_smoker_percentage,'\t',female_smoker_percentage)

23.5207100591716 	 17.371601208459214


patient number from one region

In [27]:
regions = mi.groupby(['region'])['region'].count()
print(regions)

region
northeast    324
northwest    325
southeast    364
southwest    325
Name: region, dtype: int64


average age

In [30]:
av_age = mi.groupby(['sex'])['age'].mean()
av_age

sex
female    39.503021
male      38.917160
Name: age, dtype: float64

average charge

In [32]:
av_charge = mi.groupby(['sex'])['charges'].mean()
av_charge

sex
female    12569.578844
male      13956.751178
Name: charges, dtype: float64

average bmi

In [33]:
av_bmi = mi.groupby(['sex'])['bmi'].mean()
av_bmi

sex
female    30.377749
male      30.943129
Name: bmi, dtype: float64

average age of people with at least 1 child

In [35]:
av_age_with_children = (mi.loc[mi['children'] >= 1, 'age']).mean()
av_age_with_children

39.78010471204188

Average female age(39.50) and average age with children (39.78) are very similar which means the dataset is probably based on middle aged family-oriented members.
The individuals distibution by region is almost equal (324,325,364,325), with slightly higher participants from the southeast (364)
There is a 6 % deficit between the male and the female smokers (23.5 and 17.4)
The male individuals pay more for insurance (male:13956.75, female:12569.58) 
This may be caused by the fact that the male smokers take bigger part from the dataset.

In [38]:
av_charge_non_smoker_male = mi.loc[(mi['sex'] == 'male') & (mi['smoker'] == 'no'), 'charges'].mean()
av_charge_non_smoker_female = mi.loc[(mi['sex'] == 'female') & (mi['smoker'] == 'no'), 'charges'].mean()
av_charge_smoker_male = mi.loc[(mi['sex'] == 'male') & (mi['smoker'] == 'yes'), 'charges'].mean()
av_charge_smoker_female = mi.loc[(mi['sex'] == 'female') & (mi['smoker'] == 'yes'), 'charges'].mean()

av_charge_non_smoker_male, av_charge_non_smoker_female, av_charge_smoker_male, av_charge_smoker_female


(8087.204731276595, 8762.297299541133, 33042.00597528303, 30678.996276260867)

The group of female non-smokers pay more (8762.30) than the males (8087.20) and the smoker males pay much more (33042.01) than the smoker females (30679.00)